CorsixTH/Src/th_gfx_sdl.cpp

```diff
@@ -82,6 +82,66 @@ static inline uint32_t makeSwapRedBlue(uint8_t iOpacity, uint8_t iR, uint8_t iG,
     return palette::pack_argb(iOpacity, iNewRed, iG, static_cast<uint8_t>(iNewBlue));
 }
 
+palette::palette()
+{
+    colour_count = 0;
+}
+
+static constexpr uint8_t gs_iTHColourLUT[0x40] = {
+    // Maps 0-63 to 0-255
+    0x00, 0x04, 0x08, 0x0C, 0x10, 0x14, 0x18, 0x1C,
+    0x20, 0x24, 0x28, 0x2D, 0x31, 0x35, 0x39, 0x3D,
+    0x41, 0x45, 0x49, 0x4D, 0x51, 0x55, 0x59, 0x5D,
+    0x61, 0x65, 0x69, 0x6D, 0x71, 0x75, 0x79, 0x7D,
+    0x82, 0x86, 0x8A, 0x8E, 0x92, 0x96, 0x9A, 0x9E,
+    0xA2, 0xA6, 0xAA, 0xAE, 0xB2, 0xB6, 0xBA, 0xBE,
+    0xC2, 0xC6, 0xCA, 0xCE, 0xD2, 0xD7, 0xDB, 0xDF,
+    0xE3, 0xE7, 0xEB, 0xEF, 0xF3, 0xF7, 0xFB, 0xFF,
+};
```

Comment:
LUTs are a bit obsolete with modern processors, you may want to use the `color` function that I wrote for this:
``` cpp
static unsigned char colors[] = {
    0x00, 0x04, 0x08, 0x0C, 0x10, 0x14, 0x18, 0x1C,
    0x20, 0x24, 0x28, 0x2D, 0x31, 0x35, 0x39, 0x3D,
    0x41, 0x45, 0x49, 0x4D, 0x51, 0x55, 0x59, 0x5D,
    0x61, 0x65, 0x69, 0x6D, 0x71, 0x75, 0x79, 0x7D,
    0x82, 0x86, 0x8A, 0x8E, 0x92, 0x96, 0x9A, 0x9E,
    0xA2, 0xA6, 0xAA, 0xAE, 0xB2, 0xB6, 0xBA, 0xBE,
    0xC2, 0xC6, 0xCA, 0xCE, 0xD2, 0xD7, 0xDB, 0xDF,
    0xE3, 0xE7, 0xEB, 0xEF, 0xF3, 0xF7, 0xFB, 0xFF,
};

/*
def f(i):
    return int(i*4.048 + 0.5)
*/

static unsigned char color(int index) {
    return static_cast<int>(index * 4.048 + 0.5);
}

#include <cstdio>
#include <cstdlib>

int main() {
    for (int i = 0; i <64; i++) {
        int t = colors[i];
        int c = color(i);
        if (t == c) continue;
        printf("%d: table=%d, func=%d\n", i, t, c);
    }
    return 0;
}
```
Complete with test, doesn't print any output at my system (both with the Python `f` and the cpp `color` functions).

Reply:
Sorry, being silly, the better function is of course
``` cpp
return static_cast<int>(255 * index / 63.0 + 0.5);
```

Reply:
I just moved this without looking too closely at how it was used. Will consider.

Reply:
Agree, this isn't even a performance sensitive path.